.github/workflows/ci.yml

```
@@ -0,0 +1,51 @@
+name: CI
+on:
+  pull_request:
+    branches:
+      - master
+  push:
+    branches:
+      - master
+    tags: '*'
+jobs:
+  test:
+    name: Julia ${{ matrix.version }} - ${{ matrix.os }} - ${{ matrix.arch }} - ${{ github.event_name }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        arch:
+          - x64
+        os:
+          - ubuntu-latest
+        version:
+          # - '1.0'
+          # - '1.1'
+          - '1.2'
+          - '1.3'
+          - '1.4'
+          - '1.5'
+          - '1' # automatically expands to the latest stable 1.x release of Julia
+          - 'nightly'
```

When the test passes, we can maybe switch to: (windows and macOS CI resources are quite expensive and I think it's enough to only have 1.x test on these).

```suggestion
      matrix:
        version: ['1.2', '1', 'nightly']
        os: [ubuntu-latest]
        arch: [x64]
        include:
          - os: windows-latest
            version: '1'
            arch: x64
          - os: macOS-latest
            version: '1'
            arch: x64
```

---

Yes, I agree, it isn't necessary to cover all the version and burn tons of CI credits.

---

The idea of my matrix setup is to:

- provide a test on lower-bound version `1.2` (as we declared in `Project.toml`). I believe this is necessary since it usually catches many unexpected compatibility bugs.
- only test windows and macOS platform on Julia 1.x version.
- `1.2` and `nightly` is only tested on Ubuntu.

This should be sufficient enough to catch most issues while use as little window/macOS CI credits as possible.

---

Thanks for the explanation, I will change that.